IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -173,4 +185,90 @@ For asymmetric keys this holds the public key hash which allows digital identity
+    NSData *tagData = [kMSIDPrivateKeyIdentifier dataUsingEncoding:NSUTF8StringEncoding];
+    // Legacy registrations would have be done using RSA, passing keyType = RSA in query
+    NSDictionary *extraPrivateKeyAttributes = @{ (__bridge id)kSecAttrApplicationTag: tagData,
+                                                 (__bridge id)kSecAttrAccessGroup : legacySharedAccessGroup,
+                                                 (__bridge id)kSecAttrKeyType : (__bridge id)kSecAttrKeyTypeRSA };
+    NSDictionary *extraCertAttributes = @{ (__bridge id)kSecAttrAccessGroup : legacySharedAccessGroup };
+    
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"Checking Legacy keychain for registration.");
+    MSIDWPJKeyPairWithCert *legacyKeys = [self findWPJRegistrationInfoWithAdditionalPrivateKeyAttributes:extraPrivateKeyAttributes certAttributes:extraCertAttributes context:context];
+        
+    if (legacyKeys)
+    {
+        if ([NSString msidIsStringNilOrBlank:tenantId])
+        {
+            // ESTS didn't request a specific tenant, just return default one
+            return legacyKeys;
+        }
+        
+        // Read tenantId for legacy identity
+        NSError *tenantIdError = nil;
+        NSString *registrationTenantId = [MSIDWorkPlaceJoinUtil getWPJStringDataForIdentifier:kMSIDTenantKeyIdentifier context:context error:&tenantIdError];
+        
+        // There's no tenantId on the registration, or it mismatches what server requested, keep looking for a better match. Otherwise, return the identity already.
+        if (!tenantIdError
+            && registrationTenantId
+            && [registrationTenantId isEqualToString:tenantId])
+        {
+            return legacyKeys;
+        }
+    }
+    
+    if (!tenantId)
+    {
+        // default registration should have a tenantId associated.
+        return legacyKeys;
+    }
+    MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"Checking keychain for default registration done using RSA key.");
+    NSString *defaultSharedAccessGroup = [NSString stringWithFormat:@"%@.com.microsoft.workplacejoin.v2", teamId];
+    NSString *tag = [NSString stringWithFormat:@"%@#%@", kWPJPrivateKeyIdentifier, tenantId];
+    tagData = [tag dataUsingEncoding:NSUTF8StringEncoding];
+    // Default registrations can be done using RSA/ECC. 1st Looking for RSA device key in the keychain.
+    extraPrivateKeyAttributes = @{ (__bridge id)kSecAttrApplicationTag : tagData,
+                                   (__bridge id)kSecAttrAccessGroup : defaultSharedAccessGroup,
+                                   (__bridge id)kSecAttrKeyType : (__bridge id)kSecAttrKeyTypeRSA };
+    
+    extraCertAttributes = @{ (__bridge id)kSecAttrAccessGroup : defaultSharedAccessGroup };
+    
+    MSIDWPJKeyPairWithCert *defaultKeys = [self findWPJRegistrationInfoWithAdditionalPrivateKeyAttributes:extraPrivateKeyAttributes certAttributes:extraCertAttributes context:context];
```

Comment:
Should this query be disabled for macOS?

Reply:
Yep, refactored code to not check for default registration + RSA on macOS